### ISSUE TITLE
error composer install after git pull

### ISSUE THREAD

**Issue by michaelpporter** [acquia/lightning-project#6]
Generating autoload files

  [RuntimeException]
  Could not scan for classes inside "scripts/ScriptHandler.php" which does not appear to be a file nor a folder

install [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--no-plugins] [--no-custom-installers] [--no-autoloader] [--no-scripts] [--no-progress] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--ignore-platform-reqs] [--] [<packages>]...

=================

Steps to repeat:

composer create-project acquia/lightning-project:8.1.* xmd8 --no-interaction --stability rc
cd xmd8
git init
edit .gitignore # exclude tmp*
git add .
git add remote....
git push....
cd ..
rm xmd8
git clone xmd8 repo
cd xmd8
composer install

**Comment by michaelpporter** [comment]
ps this is running in a VM, working on a setup that will work after a vagrant destroy.

**Comment by phenaproxima** [comment]
I'm unable to reproduce this problem. I wasn't doing it in a VM, but it works for me with, more or less, the steps you provide. Can you give more information about the VM itself? Was it a drupalvm instance?

**Comment by michaelpporter** [comment]
yes, it'd DrupalVM

shortened steps:
vagrant up
vagrant ssh
cd /var/www (mapped to ~/vagrant/xmd8/www)
[sudo] composer create-project... (I have tried both sudo and as vagrant)
exit
git commit (vm and project is one repo)
vagrant destroy
rm ~/vagrant/xmd8
git clone xmd8
vagrant up
cd /var/www/xmd8
composer install (both sudo and as vagrant user)
get the error above.

If I do not destroy I can run the site and install just fine. This makes me wonder if something is placed in the home folder within the VM or I should have the tmp[hash] in git. I have not tested this thought yet.


~/vmpath/.gitignore

```
.DS_Store
.idea/
.project/
.vagrant/
.bundle/
*.retry
vagrant_ansible_inventory_default
config.yml
drupal.make.yml
Vagrantfile.local
examples/prod/inventory
examples/prod/bootstrap/vars.yml
scripts/
roles/
www/html/
www/dashboard/
```


~/vmpath/www/xmd8/.gitignore
```
# Ignore binaries.
/bin

# Ignore non-custom parts of Drupal.
/docroot/core
/docroot/modules/contrib
/docroot/profiles/contrib
/docroot/themes/contrib
/docroot/sites/default/files

# Ignore Composer-managed dependencies.
vendor/

tmp*

```

**Comment by michaelpporter** [comment]
key config.yml settings

```
vagrant_synced_folders:
  # The first synced folder will be used for the default Drupal installation, if
  # build_makefile: is 'true'.
  - local_path: ~/vagrant/xmd8/www
    destination: /var/www
    type: nfs
    create: true

drupalvm_webserver: apache

build_makefile: false

install_site: false

drupal_core_path: "/var/www/xmd8/docroot"
```

**Comment by phenaproxima** [comment]
I have a theory about this -- I suspect it might be NFS being fiddly.

My recommendation is to try running the ```composer create-project``` command from *outside* the VM, not from within it.

**Comment by michaelpporter** [comment]
I can try that again; I get php timezone errors since I am only using the VM now and stopped running apache on my mac.

**Comment by phenaproxima** [comment]
If you set your timezone in /etc/php.ini on your Mac, it will go away ;)

**Comment by michaelpporter** [comment]
For those using homebrew 
```
php --ini
Loaded Configuration File:         /usr/local/etc/php/5.6/php.ini
```
Same error:

New VM (same settings)
cd www
composer create-project ...
git add .
git push
remove the VM
git clone the vm
cd vm/www
composer install

[RuntimeException]
  Could not scan for classes inside "scripts/ScriptHandler.php" which does not appear to be a file nor a folder


============

grrr the vm is ignoring "scripts/"
https://github.com/geerlingguy/drupal-vm/blob/master/.gitignore

added this to fix it.
`!www/*/scripts/`

**Comment by balsama** [comment]
@michaelpporter Sounds like this should be closed?

**Comment by michaelpporter** [comment]
Agreed.

**Comment by kieranacquia** [comment]
I installed lightning_df locally on my mac after building it using buildsteps. I checkedout the git repo master-build locally and then ran composer update to update the composer.lock file.

I got:
Could not scan for classes inside "scripts/ScriptHandler.php" which does not appear to be a file nor a   
  folder 

Since it was local on Macbook I don't think this is related to the infrastructure but rather some dependencies.
